### PR TITLE
Add support for spot instances in node groups

### PR DESCRIPTION
**Pull request by marchelleboid**
You can specify multiple instance types with spot instances, but we'll save that for a future PR.

If you get an error about not being able to use `--node-security-groups` with a managed node group, try upgrading `eksctl`.

[sc-78511]

### CODE REVIEW

**Review by NawreCode — APPROVED** [review]
LGTM. But make sure to wait for sc 80994 to be merged before merging this one, because updating `eksctl` is a pain.

**Review by NawreCode — APPROVED** [review]

**Review by nikopol — APPROVED** [review]
LGTM. test with DSS 10.0.5, but like Erwan, I had to update `eksctl`.

**Pull request merged**
